Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/AssetBrowserTreeToTableProxyModel.cpp

```diff
@@ -9,1008 +9,162 @@
-                    });
-
-                connect(
-                    model,
-                    &QAbstractItemModel::rowsMoved,
-                    this,
-                    [this](
-                        const QModelIndex& srcParent,
-                        int srcStart,
-                        [[maybe_unused]] int srcEnd,
-                        const QModelIndex& destParent,
-                        int destStart)
-                    {
-                        RowsMoved(srcParent, srcStart, destParent, destStart);
-                    });
-
-                connect(
-                    model,
-                    &QAbstractItemModel::modelAboutToBeReset,
-                    this,
-                    [this]()
-                    {
-                        beginResetModel();
-                        resetInternalData();
-                        m_rowCount = 0;
-                        m_map.Clear();
-                        m_parents.clear();
-                    });
-
-                connect(
-                    model,
-                    &QAbstractItemModel::modelReset,
-                    this,
-                    [this]()
-                    {
-                        ModelReset();
-                    });
-
-                connect(
-                    model,
-                    &QAbstractItemModel::dataChanged,
-                    this,
-                    [this](const QModelIndex& topLeft, const QModelIndex& bottomRight)
-                    {
-                        DataChanged(topLeft, bottomRight);
-                    });
-
-                connect(
-                    model,
-                    &QAbstractItemModel::layoutAboutToBeChanged,
-                    this,
-                    [this]()
-                    {
-                        LayoutAboutToBeChanged();
-                    });
-
-                connect(
-                    model,
-                    &QAbstractItemModel::layoutChanged,
-                    this,
-                    [this]()
-                    {
-                        LayoutChanged();
-                    });
-
-                connect(
-                    model,
-                    &QObject::destroyed,
-                    this,
-                    [this]()
-                    {
-                        resetInternalData();
-                    });
-            }
-
-            if (model && model->hasChildren())
-            {
-                RefreshMap();
-            }
-            endResetModel();
-        }
-
-        QVariant AssetBrowserTreeToTableProxyModel::headerData(int section, Qt::Orientation orientation, int role) const
-        {
-            auto sourceModelPtr = sourceModel();
-            if (!sourceModelPtr || columnCount() <= section)
-            {
-                return QVariant();
+                    this->sourceModel(), &QAbstractItemModel::dataChanged, this, &AssetBrowserTreeToTableProxyModel::onSourceDataChanged);
```

Comment:
might be able to just a lambda and then you don't need a method just to call FlattenTree.